src/System.Private.ServiceModel/src/System/ServiceModel/DnsEndpointIdentity.cs

```diff
@@ -11,7 +16,26 @@ public DnsEndpointIdentity(string dnsName)
             if (dnsName == null)
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("dnsName");
 
-            throw ExceptionHelper.PlatformNotSupported("DnsEndpointIdentity is not supported.");
+            base.Initialize(Claim.CreateDnsClaim(dnsName));
+        }
+
+        public DnsEndpointIdentity(Claim identity)
```

Comment:
Claim is not in public contract, do we plan to expose it? Otherwise, we should remove this public Ctor.